src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java

```diff
@@ -166,6 +167,21 @@
         });
     }
 
+    @Issue("JENKINS-42952")
+    @Test public void duration() {
+        r.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = r.j.createProject(WorkflowJob.class, "this-job");
+                p.setDefinition(new CpsFlowDefinition(
+                        "echo \"currentBuild.duration='${currentBuild.duration}'\"\n" +
+                                "echo \"currentBuild.durationString='${currentBuild.durationString}'\"\n", true));
+                WorkflowRun b = r.j.assertBuildStatusSuccess(p.scheduleBuild2(0).get());
+                r.j.assertLogNotContains("currentBuild.duration='0'", b);
```

Comment:
What if it in fact took a few milliseconds to get that far? Would the test not fail?